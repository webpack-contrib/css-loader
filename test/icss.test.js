const path = require('path');
const fs = require('fs');

const { webpack, evaluated } = require('./helpers');

const testCasesPath = path.join(__dirname, 'fixtures/icss/tests-cases');
const testCases = fs.readdirSync(testCasesPath);

describe('ICSS', () => {
  testCases.forEach((name) => {
    it(`case ${name}`, async () => {
      const testId = `./icss/tests-cases/${name}/source.css`;
      const stats = await webpack(testId);
      const { modules } = stats.toJson();
      const module = modules.find((m) => m.id === testId);
      const evaluatedModule = evaluated(module.source, modules);

      expect(evaluatedModule).toMatchSnapshot('module (evaluated)');
      expect(evaluatedModule.locals).toMatchSnapshot('locals');
      expect(stats.compilation.warnings).toMatchSnapshot('warnings');
      expect(stats.compilation.errors).toMatchSnapshot('errors');
    });
  });
});
