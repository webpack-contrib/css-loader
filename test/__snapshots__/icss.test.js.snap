// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ICSS case duplicate-export: errors 1`] = `Array []`;

exports[`ICSS case duplicate-export: locals 1`] = `
Object {
  "_test": "_right_value",
}
`;

exports[`ICSS case duplicate-export: module (evaluated) 1`] = `
Array [
  Array [
    1,
    "
",
    "",
  ],
]
`;

exports[`ICSS case duplicate-export: warnings 1`] = `Array []`;

exports[`ICSS case duplicate-export-in-multiple-export: errors 1`] = `Array []`;

exports[`ICSS case duplicate-export-in-multiple-export: locals 1`] = `
Object {
  "_test": "_right_value",
}
`;

exports[`ICSS case duplicate-export-in-multiple-export: module (evaluated) 1`] = `
Array [
  Array [
    1,
    "
",
    "",
  ],
]
`;

exports[`ICSS case duplicate-export-in-multiple-export: warnings 1`] = `Array []`;

exports[`ICSS case empty-export: errors 1`] = `Array []`;

exports[`ICSS case empty-export: locals 1`] = `undefined`;

exports[`ICSS case empty-export: module (evaluated) 1`] = `
Array [
  Array [
    1,
    "
",
    "",
  ],
]
`;

exports[`ICSS case empty-export: warnings 1`] = `Array []`;

exports[`ICSS case empty-import: errors 1`] = `Array []`;

exports[`ICSS case empty-import: locals 1`] = `undefined`;

exports[`ICSS case empty-import: module (evaluated) 1`] = `
Array [
  Array [
    1,
    "
",
    "",
  ],
]
`;

exports[`ICSS case empty-import: warnings 1`] = `Array []`;

exports[`ICSS case export: errors 1`] = `Array []`;

exports[`ICSS case export: locals 1`] = `
Object {
  "_test": "_test",
}
`;

exports[`ICSS case export: module (evaluated) 1`] = `
Array [
  Array [
    1,
    "
",
    "",
  ],
]
`;

exports[`ICSS case export: warnings 1`] = `Array []`;

exports[`ICSS case import: errors 1`] = `Array []`;

exports[`ICSS case import: locals 1`] = `
Object {
  "primary-color": "red",
}
`;

exports[`ICSS case import: module (evaluated) 1`] = `
Array [
  Array [
    2,
    "
",
    "",
  ],
  Array [
    1,
    ".className {
  color: red;
}
",
    "",
  ],
]
`;

exports[`ICSS case import: warnings 1`] = `Array []`;

exports[`ICSS case multiple-export: errors 1`] = `Array []`;

exports[`ICSS case multiple-export: locals 1`] = `
Object {
  "_foo": "_bar",
  "_test": "_test",
}
`;

exports[`ICSS case multiple-export: module (evaluated) 1`] = `
Array [
  Array [
    1,
    "
",
    "",
  ],
]
`;

exports[`ICSS case multiple-export: warnings 1`] = `Array []`;

exports[`ICSS case multiple-keys-values-in-export: errors 1`] = `Array []`;

exports[`ICSS case multiple-keys-values-in-export: locals 1`] = `
Object {
  "_test": "_test",
  "_test1": "1",
  "_test2": "'string'",
  "_test3": "1px 2px 3px",
  "_test4": "1px 2px 3px, 1px 2px 3px",
}
`;

exports[`ICSS case multiple-keys-values-in-export: module (evaluated) 1`] = `
Array [
  Array [
    1,
    "
",
    "",
  ],
]
`;

exports[`ICSS case multiple-keys-values-in-export: warnings 1`] = `Array []`;
